src/app/teacher/select-runs-controls/select-runs-controls.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MAT_CHECKBOX_DEFAULT_OPTIONS } from '@angular/material/checkbox';
+import { TeacherRun } from '../teacher-run';
+import { ArchiveProjectResponse } from '../../domain/archiveProjectResponse';
+import { Subscription } from 'rxjs';
+import { Project } from '../../domain/project';
+import { ArchiveProjectService } from '../../services/archive-project.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'select-runs-controls',
+  templateUrl: './select-runs-controls.component.html',
+  styleUrls: ['./select-runs-controls.component.scss'],
+  providers: [{ provide: MAT_CHECKBOX_DEFAULT_OPTIONS, useValue: { clickAction: 'noop' } }]
+})
+export class SelectRunsControlsComponent {
+  @Output() archiveActionEvent = new EventEmitter<void>();
+  protected numSelectedRuns: number = 0;
+  @Input() runChangedEventEmitter: EventEmitter<void> = new EventEmitter<void>();
+  @Input() runs: TeacherRun[] = [];
+  protected selectedAllRuns: boolean = false;
+  protected selectedSomeRuns: boolean = false;
+  @Output() selectRunsOptionChosenEvent = new EventEmitter<string>();
+  @Input() showArchived: boolean = false;
+
+  constructor(
+    private archiveProjectService: ArchiveProjectService,
+    private snackBar: MatSnackBar
+  ) {}
+
+  ngOnInit(): void {
+    this.runChangedEventEmitter.subscribe(() => {
+      this.ngOnChanges();
+    });
+  }
+
+  ngOnChanges(): void {
+    this.numSelectedRuns = this.runs.filter((run: TeacherRun) => run.selected).length;
+    this.selectedAllRuns = this.numSelectedRuns === this.runs.length;
+    this.selectedSomeRuns = this.numSelectedRuns !== 0 && !this.selectedAllRuns;
+  }
+
+  protected selectAllRunsCheckboxClicked(): void {
+    this.selectRunsOptionChosenEvent.emit(
+      this.selectedAllRuns || this.selectedSomeRuns ? 'none' : 'all'
+    );
+  }
+
+  protected selectRunsOptionChosen(value: string): void {
+    this.selectRunsOptionChosenEvent.emit(value);
+  }
```

Comment:
Emit SelectRunsOption enum value?
```suggestion
  protected selectAllRunsCheckboxClicked(): void {
    this.selectRunsOptionChosenEvent.emit(
      this.selectedAllRuns || this.selectedSomeRuns ? SelectRunsOption.None : SelectRunsOption.All
    );
  }

  protected selectRunsOptionChosen(value: string): void {
    this.selectRunsOptionChosenEvent.emit(value as SelectRunsOption);
  }
```